runtime/src/tiered_storage/footer.rs

```diff
@@ -0,0 +1,352 @@
+use {
+    crate::tiered_storage::{file::TieredStorageFile, mmap_utils::get_type},
+    memmap2::Mmap,
+    serde::{Deserialize, Serialize},
+    solana_sdk::hash::Hash,
+    std::{mem, path::Path},
+};
+
+pub const FOOTER_FORMAT_VERSION: u64 = 1;
+
+// The size of the footer struct + the u64 magic number at the end.
+pub const FOOTER_SIZE: i64 = (mem::size_of::<TieredStorageFooter>() + mem::size_of::<u64>()) as i64;
```

Comment:
Because we will be seeking backward using the FOOTER_SIZE, and SeekFrom::End takes i64 instead of u64.

Or I can delay the casting before the function call.